### PR TITLE
Update config format

### PR DESCRIPTION
**Pull request by colinhoglund**
This updates the CLI to use a configuration file Instead of CLI flags, which makes the extension router more flexible for adding new plugins.

Changes:
- Switch from `viper` to `envconfig`, which allows for a simplified configuration/environment loading process.
- Adds a new `internal/config` package which contains the structs for unmarshalling the configuration file and methods for retrieving enabled plugins.
- Updated readme with usage instructions and a plugin index

### CODE REVIEW

File: internal/cli/cli.go [hunk 1]
```diff
@@ -1,79 +1,66 @@
...
-	if err != nil {
-		return err
-	}
+	if c.ConfigFile != "" {
+		data, err := os.ReadFile(c.ConfigFile)
+		if err != nil {
+			return err
+		}
 
-	log.SetLevel(logLevel)
```
Comment (whwalter) [inline internal/cli/cli.go:52]: Seems setting the log level gets dropped in the config changes.

**Review by whwalter — COMMENTED** [review]
I think log level needs to be set still but the rest looks good. I had a suggestion for a more generic config parser but what you have works.

File: internal/cli/cli.go (hunk 1 above)
Comment (colinhoglund) [reply internal/cli/cli.go:52]: fixed

File: internal/config/config.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		Convert Convert `yaml:"convert"`
+	}
+
+	Convert struct {
+		Pathschanged Pathschanged `yaml:"pathschanged"`
+	}
+
+	Pathschanged struct {
+		Enable bool `yaml:"enable"`
+	}
```
[the remark this answers is not in the document]
Comment (colinhoglund) [reply internal/config/config.go:19]: I tried something like this, but opted for something a bit simpler for now for the sake of readability. This is something we should be able to implement pretty easily if needed.

**Review by whwalter — APPROVED** [review]

**Pull request merged**